tests/storage/test_room_search.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import synapse.rest.admin
+from synapse.rest.client import login, room
+from synapse.storage.engines import PostgresEngine
+
+from tests.unittest import HomeserverTestCase
+
+
+class NullByteInsertionTest(HomeserverTestCase):
+    servlets = [
+        synapse.rest.admin.register_servlets_for_client_rest_resource,
+        login.register_servlets,
+        room.register_servlets,
+    ]
+
+    def test_null_byte(self):
+        """
+        Postgres/SQLite don't like null bytes going into the search tables. Internally
+        we replace those with a space.
```

Comment:
I thought this was only a Postgres problem?

Reply:
see my comment above: 
> -testing revealed that while SQLite doesn't complain when you insert a text string with "\u0000", it will stop processing that string at null byte when queried later. This causes search to fail silently when searching for the text that was inserted into the `events_search` table containing the null byte. While minor, I figured this was not ideal so I modified the code so that we are also substituting space for "\u0000" when inserting into a SQLite version of `events_search`.

Reply:
Sorry, I missed that!